src/Language/Fixpoint/Solver/PLE.hs

```diff
@@ -1154,42 +1161,36 @@ instance Normalizable Equation where
 normalizeBody :: Symbol -> Expr -> Expr
 normalizeBody f = go
   where
-    go e
-      | f `elem` syms e
-      = go' e
-    go e
-      = e
-
-    go' (PAnd [PImp c e1,PImp (PNot c') e2])
-      | c == c' = EIte c e1 (go' e2)
-    go' e = e
-
-_splitBranches :: Symbol -> Expr -> [(Expr, Expr)]
-_splitBranches f = go
-  where
-    go (PAnd es)
-      | f `elem` syms es
-      = go' <$> es
-    go e
-      = [(PTrue, e)]
+    -- Q: Why do we only simplify the expression if f is a free symbol in it?
```

Comment:
Question 1: Why do we only simplify the expression if f is a free symbol in it?

Reply:
`normalizeBody f bd` is  only called when `bd` is the definition of `f`. We only need to normalize when `f` is recursive. So, we have this check for efficiency: I believe nothing bad will happen if you normalize anyways, but you do not need to.

Reply:
Can we add @nikivazou 's remark to answer the question in the comment?

Reply:
I think this has be done in the last commit: eaff89e